captain-definition

```diff
@@ -0,0 +1,15 @@
+{
+ "schemaVersion" :2 ,
+ "dockerfileLines" :[
+                "FROM python:3.7.0-alpine",
+                "COPY ./ /app",
+                "WORKDIR /app",
+                "RUN ls -a",
+                "RUN apk add --no-cache gcc libc-dev libxslt-dev postgresql-dev",
+                "RUN pip3 install --upgrade pip",
+                "RUN pip3 install -r requirements.txt",
+                "RUN pip3 install email-validator",
```

Comment:
why install email-validator?

Reply:
fails to run if i don't install it:
```
2020-06-23T04:55:21.924011045Z Traceback (most recent call last):
2020-06-23T04:55:21.924186575Z File "app.py", line 10, in <module>
2020-06-23T04:55:21.924193696Z from flask_security.utils import encrypt_password
2020-06-23T04:55:21.924196727Z File "/usr/local/lib/python3.7/site-packages/flask_security/__init__.py", line 13, in <module>
2020-06-23T04:55:21.924201520Z from .core import Security, RoleMixin, UserMixin, AnonymousUser, current_user
2020-06-23T04:55:21.924204599Z File "/usr/local/lib/python3.7/site-packages/flask_security/core.py", line 25, in <module>
2020-06-23T04:55:21.924207547Z from .forms import LoginForm, ConfirmRegisterForm, RegisterForm, \
2020-06-23T04:55:21.924210303Z File "/usr/local/lib/python3.7/site-packages/flask_security/forms.py", line 68, in <module>
2020-06-23T04:55:21.924213252Z email_validator = Email(message='INVALID_EMAIL_ADDRESS')
2020-06-23T04:55:21.924215976Z File "/usr/local/lib/python3.7/site-packages/wtforms/validators.py", line 332, in __init__
2020-06-23T04:55:21.924218765Z raise Exception("Install 'email_validator' for email validation support.")
2020-06-23T04:55:21.924221627Z Exception: Install 'email_validator' for email validation support.
2020-06-23T04:55:28.528063652Z Traceback (most recent call last):
2020-06-23T04:55:28.528158696Z File "app.py", line 10, in <module>
2020-06-23T04:55:28.528163123Z from flask_security.utils import encrypt_password
2020-06-23T04:55:28.528166014Z File "/usr/local/lib/python3.7/site-packages/flask_security/__init__.py", line 13, in <module>
2020-06-23T04:55:28.528170767Z from .core import Security, RoleMixin, UserMixin, AnonymousUser, current_user
2020-06-23T04:55:28.528173593Z File "/usr/local/lib/python3.7/site-packages/flask_security/core.py", line 25, in <module>
2020-06-23T04:55:28.528182637Z from .forms import LoginForm, ConfirmRegisterForm, RegisterForm, \
2020-06-23T04:55:28.528185419Z File "/usr/local/lib/python3.7/site-packages/flask_security/forms.py", line 68, in <module>
2020-06-23T04:55:28.528188308Z email_validator = Email(message='INVALID_EMAIL_ADDRESS')
2020-06-23T04:55:28.528191020Z File "/usr/local/lib/python3.7/site-packages/wtforms/validators.py", line 332, in __init__
2020-06-23T04:55:28.528193869Z raise Exception("Install 'email_validator' for email validation support.")
2020-06-23T04:55:28.528196703Z Exception: Install 'email_validator' for email validation support.
2020-06-23T04:55:35.112597930Z Traceback (most recent call last):
2020-06-23T04:55:35.112671825Z File "app.py", line 10, in <module>
2020-06-23T04:55:35.112676212Z from flask_security.utils import encrypt_password
2020-06-23T04:55:35.112679133Z File "/usr/local/lib/python3.7/site-packages/flask_security/__init__.py", line 13, in <module>
2020-06-23T04:55:35.112683561Z from .core import Security, RoleMixin, UserMixin, AnonymousUser, current_user
2020-06-23T04:55:35.112686381Z File "/usr/local/lib/python3.7/site-packages/flask_security/core.py", line 25, in <module>
2020-06-23T04:55:35.112689383Z from .forms import LoginForm, ConfirmRegisterForm, RegisterForm, \
2020-06-23T04:55:35.112692144Z File "/usr/local/lib/python3.7/site-packages/flask_security/forms.py", line 68, in <module>
2020-06-23T04:55:35.112695118Z email_validator = Email(message='INVALID_EMAIL_ADDRESS')
2020-06-23T04:55:35.112697812Z File "/usr/local/lib/python3.7/site-packages/wtforms/validators.py", line 332, in __init__
2020-06-23T04:55:35.112700609Z raise Exception("Install 'email_validator' for email validation support.")
2020-06-23T04:55:35.112703543Z Exception: Install 'email_validator' for email validation support.
2020-06-23T04:55:41.734896107Z Traceback (most recent call last):
2020-06-23T04:55:41.734993909Z File "app.py", line 10, in <module>
2020-06-23T04:55:41.734999300Z from flask_security.utils import encrypt_password
2020-06-23T04:55:41.735002423Z File "/usr/local/lib/python3.7/site-packages/flask_security/__init__.py", line 13, in <module>
2020-06-23T04:55:41.735006866Z from .core import Security, RoleMixin, UserMixin, AnonymousUser, current_user
2020-06-23T04:55:41.735009671Z File "/usr/local/lib/python3.7/site-packages/flask_security/core.py", line 25, in <module>
2020-06-23T04:55:41.735012625Z from .forms import LoginForm, ConfirmRegisterForm, RegisterForm, \
2020-06-23T04:55:41.735015321Z File "/usr/local/lib/python3.7/site-packages/flask_security/forms.py", line 68, in <module>
2020-06-23T04:55:41.735018186Z email_validator = Email(message='INVALID_EMAIL_ADDRESS')
2020-06-23T04:55:41.735020922Z File "/usr/local/lib/python3.7/site-packages/wtforms/validators.py", line 332, in __init__
2020-06-23T04:55:41.735023717Z raise Exception("Install 'email_validator' for email validation support.")
2020-06-23T04:55:41.735026530Z Exception: Install 'email_validator' for email validation support.
```

Reply:
looks like this was removed in the latest? version of wtforms. Not sure why even if you're not using it, its still required to be installed. One other solution is to use an earlier version of wtforms. reverting back "wtforms.2.2.1" seems to fix the issue.

Reply:
ok, well better to add it to requirements.txt so that all forms of installation can benefit

Reply:
> ok, well better to add it to requirements.txt so that all forms of installation can benefit

install email-validator or install earlier version of wtforms?

Reply:
just add email-validator

Reply:
added it and pushed.